modules/gateway/peers.go

```diff
@@ -348,6 +403,45 @@ func acceptConnVersionHandshake(conn net.Conn, version string) (remoteVersion st
 	return remoteVersion, nil
 }
 
+// acceptConnSessionHandshake performs the session handshake and should be
+// called on the side accepting a connection request.
+// if err == nil, wantConn indicates if a connection is wanted on both sides.
+func acceptConnSessionHandshake(conn net.Conn, header sessionHeader) (wantConn bool, err error) {
+	var remoteHeader sessionHeader
+	// Read remote header.
+	if err = encoding.ReadObject(conn, &remoteHeader, build.MaxEncodedVersionLength); err != nil {
```

Comment:
we need a separate constant for the maximum length. It should be `sizeof(BlockID) + sizeof(gatewayID) + sizeof(bool)` == `32 + 8 + 1` = `41`. Define it in `gateway/consts.go`